### DISCUSSION TITLE
How to end drag if mouse moves outside of carousel container?

### DISCUSSION THREAD

**Discussion by hanshank** [davidjerleke/embla-carousel#497] [Help]
My team is rebuilding a `slick` based carousel using Embla and we are trying to match the spec. Curious if it is possible with the API or options to automatically scroll to the next snap when the user starts a drag and eventually moves their mouse outside of the carousel container? Currently I can drag indefinitely, as long as there is space on my monitor, without the slides snapping into place. Sorta hard to explain, but please let me know if I need to clarify

**Comment by davidjerleke** [comment]
Hi @hanshank,

I'm not sure there are any obvious benefits mimicking slick here. Here's why in my opinion:

- Not tracking the pointer when it has moved outside of the element bounds (assuming the pointer down event happened inside the element), we might miss important changes in how fast the user is dragging the carousel.

- Some use cases may be carousels that doesn't take up the full width/height of the window. In these cases only tracking the pointer inside the carousel element would make it hard to drag in a predictable way, if the slightest drag outside the element bounds were to stop tracking the pointer (because the carousel has a small area on the webpage, its draggable area would also be small).

- How do we know if dragging the pointer outside of the element was intentional? If I'm dragging the carousel with a bit of force, but don't intend to actually move the cursor outside of the carousel element, the scroll speed wouldn’t be proportional to the drag force.

To sum up, I don't think this should be prioritized right now among other things that could be improved with Embla Carousel. I hope this makes sense.

Best
David
